src/evergreen-updater.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+'use strict'
+
+/*::
+
+export type ParseUpdateOptions = {
+  data: {
+    available: boolean,
+    compatible: boolean,
+    snapshot: string,
+    url: string
+  }
+}
+
+export type IonicDeploy = {
+  download: (string, Function, Function) => void,
+  redirect: (string) => void,
+  parseUpdate: (string, ParseUpdateOptions, Function, Function) => void,
+  deploy: (string, Function, Function) => void,
+}
+
+*/
+
+export default class EvergreenUpdater {
+  /*:: ionicDeploy: any */
+  /*:: appId: string */
+  /*:: tennant: string */
+  /*:: zipUrl: string */
+
+  constructor (
+    ionicDeploy /*: any */,
+    appId /*: string */,
+    tennant /*: string */,
+    platformId /*: string */
+  ) /*:void */ {
+    if (!ionicDeploy) {
+      throw new Error('ionicDeploy Plugin is missing')
+    }
+
+    if (!appId) {
+      throw new Error('appId was not specified')
+    }
+
+    if (!tennant) {
+      throw new Error('tennant was not specified')
+    }
+
+    if (!platformId) {
+      throw new Error('platformId was not specified')
+    }
+
+    this.ionicDeploy = ionicDeploy
+    this.appId = appId
+    this.tennant = tennant
+    this.zipUrl = `https://evergreen.blinkm.io/${this.tennant}/${this.appId}/www-${platformId}.zip`
+
+    this.ionicDeploy.init(this.appId, 'https://evergreen.blinkm.io/fake/deploy')
+  }
+
+  check () /*: Promise<boolean> */ {
+    return this._getEtag()
+      .then((etag) => this._parseUpdate(etag))
+      .catch(() => false)
+  }
+
+  download (onProgressCb /*: function */ = () => true) {
```

Comment:
This annotation should prob be `/*: ?function */`. The `?` signifies that the argument may not be passed in at all.